flang/lib/Lower/CharacterExpr.cpp

```diff
@@ -33,8 +35,11 @@ static fir::CharacterType getCharacterType(const fir::CharBoxValue &box) {
 }
 
 static bool needToMaterialize(const fir::CharBoxValue &box) {
-  return box.getBuffer().getType().isa<fir::SequenceType>() ||
-         box.getBuffer().getType().isa<fir::CharacterType>();
+  auto buffTy = box.getBuffer().getType();
```

Comment:
This function needs to be revisited/renamed, it was mainly intended as "Is this thing in memory" ? "Can we load from / assign to it ? ". All that to say, I am not sure I understand the change here, but that's fine since I want to get rid of it at some point anyway.